blog/_posts/2022-02-04-policy-objects-routing-example.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "Policy Objects: How to configure firewalld for routed networks in libvirt"
+section: Blog
+date: 2022-02-04T15:00:00+1100
+author: Benjamin Blasco
+category: feature
+---
+
+## Introduction
+
+If you're reading this you may have searched the web for "libvirt routed firewalld" or something similar. 
+
+This article is all about enabling the following configuration on a Red Hat Enterprise Linux/CentOS/Fedora or similar host:
+- VMs running in your machine with libvirt
+- Routed networks for the VMs (rather than the default NAT setup libvirt uses)
+- Firewalld with NFTables as the back-end to protect your host and guests
+
+This seems like it should all be pretty easy, right?  Routed networks with libvirt are relatively straightforward by themselves, and firewalld is also straightforward to configure.  Both are quite well documented, however I couldn't get them to work together.
+
+Here's my desired configuration to route traffic from my workstation (laptop) and my VMs hosted on my hypervisor host:
+![Network configuration for routed libvirt network](2022-02-04-routed-networks-libvirt-firewalld-v2.png)
+
+## What's the challenge?
+
+After much research and troubleshooting I learned that firewalld was rejecting incoming ping and SSH packets, with error messages in the logs (which appear in /var/log/messages and the output of `dmesg` when enabled).  Here's an example of me trying to SSH from my workstation (192.168.1.99) to a VM at 192.168.6.163:
+
+```
+[Wed Jan 19 12:12:31 2022] "filter_FWD_FedoraServer_REJECT: "IN=enp2s0f0 OUT=virbr1 MAC=98:f2:b3:e9:09:98:dc:fb:48:bd:15:be:08:00 SRC=192.168.1.99 DST=192.168.6.163 LEN=60 TOS=0x08 PREC=0x40 TTL=63 ID=38395 DF PROTO=TCP SPT=49618 DPT=22 WINDOW=64240 RES=0x00 SYN URGP=0 
+```
+
+## How do we fix it?
+
+This tells me the FedoraServer zone is rejecting my request.  Why?  It turns out that the Firewalld+nftables filter_FORWARD chain prevents non-NATed routing.  The feature implementation to support is a work in progress at the time of writing this article, and more info can be found here:
+
+[\[RFE\] nftables filter_FORWARD: allow non-NATed router configurations #569](https://github.com/firewalld/firewalld/issues/569)
```

Comment:
This bug is closed. Policy objects ARE the enhancement that allows this setup to work. :)

Reply:
Corrected now.